cmd/state-svc/service.go

```diff
@@ -69,13 +68,16 @@ func (s *service) Stop() error {
 		return errs.Wrap(err, "Failed to stop server")
 	}
 
-	if err := s.ipcSrv.Close(); err != nil {
+	if err := s.ipcSrv.Shutdown(); err != nil {
 		return errs.Wrap(err, "Failed to stop ipc server")
 	}
 
 	return nil
 }
 
-func (s *service) Wait() {
-	_ = s.ipcSrv.Wait()
+func (s *service) Wait() error {
+	if err := s.ipcSrv.Wait(); err != nil {
+		return errs.Wrap(err, "IPC server operating failure")
```

Comment:
```suggestion
		return errs.Wrap(err, "Failed to wait for ipc service")
```

Error was a bit generic.

Reply:
The Start function returns errors only related to starting, Stop only related to stopping, and Wait returns errors related to normal operation. So it's not failing to wait, it's relaying a malfunction after the wait unblocks. I'm happy to include the suggestion if you still think the change is useful.